res/controllers/Pioneer-DDJ-SR2-scripts.js

```diff
@@ -0,0 +1,1600 @@
+	  const deck = DDJSR2.channelGroups[group];
+
+        if (DDJSR2.scratchMode[deck] && engine.isScratching(deck + 1)) {
+	      engine.scratchTick(deck + 1, DDJSR2.getJogWheelDelta(value));
+	  }
+    };
+
+    this.jogPlatterTick = function(channel, control, value, status, group) {
+        const deck = DDJSR2.channelGroups[group];
+	  DDJSR2.pitchBendFromJog(group, DDJSR2.getJogWheelDelta(value));
+    };
+
+    this.jogPlatterTickShift = function(channel, control, value, status, group) {
+        const deck = DDJSR2.channelGroups[group];
+
+        if (DDJSR2.scratchMode[deck] && engine.isScratching(deck + 1)) {
+	      engine.scratchTick(deck + 1, DDJSR2.getJogWheelDelta(value));
+	  } else {
+	      DDJSR2.pitchBendFromJog(
+		  group,
+		  DDJSR2.getJogWheelDelta(value) * DDJSR2.jogwheelShiftMultiplier
+	      );
+        }
+    };
+
+    this.platterTouch = function(channel, control, value, status, group) {
+	  const deck = DDJSR2.channelGroups[group];
+
+	  if (DDJSR2.scratchMode[deck]) {
+            if (value) {
+                engine.scratchEnable(
+                    deck + 1,
+                    DDJSR2.scratchSettings.jogResolution,
+                    DDJSR2.scratchSettings.vinylSpeed,
+                    DDJSR2.scratchSettings.alpha,
+                    DDJSR2.scratchSettings.beta,
+                    true
+		  );
+            } else {
+                engine.scratchDisable(deck + 1, true);
+            }
+	  }
+    };
+
+    this.vinylButton = new components.Button({
+        midi: [0x90+offset, 0x17],
+        type: components.Button.prototype.types.toggle,
+        input: function(channel, control, value, status, group) {
+            var deck = DDJSR2.channelGroups[group],
+	          midiBaseAddress = 0x90,
+                channel = 0x17,
+                midiOut = 0x00;
+
+	      if (value) {
+	          DDJSR2.scratchMode[deck] = !DDJSR2.scratchMode[deck];
+	          midi.sendShortMsg(
+                    midiBaseAddress + offset,
+                    channel,
+                    DDJSR2.scratchMode[deck]
+		  );
+            }
+        }
+    });
+	  // Defaulting the scratchmodes.
+    midi.sendShortMsg(
+        0x90 + offset,
+        0x17,
+        DDJSR2.scratchMode[offset]
+    );
+
+    // Attach the pads to the deck instance
+    this.padSection = new DDJSR2.PadSection(this, offset);
+
+
+};
+
+DDJSR2.Deck.prototype = Object.create(components.Deck.prototype);
+
+DDJSR2.crossfader = new components.Pot({
+    group: "[Master]",
+    inKey: "crossfader",
+    input: function() {
+        // We need a weird max. for the crossfader to make it cut cleanly.
+        // However, components.js resets max. to 0x3fff when the first value is
+        // received. Hence, we need to set max. here instead of within the
+        // constructor.
+        this.max = (0x7f<<7) + 0x70;
+        components.Pot.prototype.input.apply(this, arguments);
+    }
```

Comment:
please elaborate. Why couldn't you just do:
```suggestion
    max: (0x7f<<7) + 0x70,
```
Also why the exactly the weird value, what values does the controller send when you put the crossfader to the right?